packages/daimo-common/src/request.ts

```diff
@@ -18,3 +28,41 @@ export function generateRequestId() {
 
   return hexToBigInt(hexSeed);
 }
+
+/**
+ * Handle rich request metadata parsing.
+ * In the future, we may consider a much stronger form of hashing here,
+ * potentially EIP-712 compliant ones.
+ */
+
+export type DaimoRichRequestV2 = {
+  v: number;
```

Comment:
add comment saying what `v` is.

i'm also kind of confused about `fulfiller`... why is this parsed metadata? the fulfiller of a request is whoever paid it according to onchain event log

Reply:
It's a little bit of a misnomer, probably should be something like `proposedFulfiller`. It refers to the user to whom the request is sent. Since a request can be fulfilled by anyone, there is no record of the request receiver onchain, so it is added to the request metadata.